src/Vibrato/Vibrato.cpp

```diff
@@ -0,0 +1,129 @@
+#include "Vibrato.h"
+#include "Lfo.h"
+
+CVibrato::CVibrato():
+    m_bIsInitialized(false),
+    m_fSampleRate(0),
+    m_fDelayTimeInS(0),
+    m_fVibratoRangeInS(0),
+    m_fVibratoFrequencyInHz(0)
+{
+    this->reset();
+}
+
+CVibrato::~CVibrato()
+{
+    this->reset();
+}
+
+Error_t CVibrato::create(CVibrato*& pCVibrato)
+{
+    pCVibrato = new CVibrato;
+    if (!pCVibrato)
+        return Error_t::kMemError;
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::destroy(CVibrato*& pCVibrato)
+{
+    if (pCVibrato->m_bIsInitialized)
+        pCVibrato->reset();
+    delete pCVibrato;
+    pCVibrato = 0;
+    return Error_t::kNoError;
+}
+
+Error_t CVibrato::init(float fDelayTimeInS, float fSampleRateInHz, int iLFOWavetableLength)
+{
+    if (m_bIsInitialized)
+    {
+        return Error_t::kFunctionIllegalCallError;
+    }
+
+    m_fSampleRate = fSampleRateInHz;
+    m_fDelayTimeInS = fDelayTimeInS;
+
+    int iMaxDelayLength = static_cast<int>(2 * std::round(m_fDelayTimeInS * m_fSampleRate)) + 2;
+
+    m_pCLFO = new CWavetableLFO(iLFOWavetableLength, m_fSampleRate, m_fVibratoFrequencyInHz, m_fVibratoRangeInS * m_fSampleRate);
+    m_pCRingBuff = new CRingBuffer<float>(iMaxDelayLength);
+    m_pCRingBuff->setReadIdx(static_cast<int>(std::round(m_fDelayTimeInS * m_fSampleRate)) + 2);
```

Comment:
So you are setting the readIdx at the end of the delay line to avoid an initial delay in the output? Or did i get this wrong, this is really interesting!

Reply:
This is a trick to set a fixed delay line. Same result could be achieved when you initialize with the write head delayed